core/src/main/java/org/mule/component/DefaultComponentLifecycleAdapter.java

```diff
@@ -339,7 +338,7 @@ public Object invoke(MuleEvent event) throws MuleException
         {
             if (componentObject == null)
             {
-                throw new ComponentException(createStaticMessage("componentObject is null"), RequestContext.getEvent(), component);
+                throw new ComponentException(createStaticMessage("componentObject is null"), event, component);
```

Comment:
@dfeist, @pablokraan I did the change to propagate the exceptionHandler from flowConstruct on AbstractMessageReceiver when the TransactionalErrorHandlingExecutionTemplate is created and reverted the change in DefaultComponentLifecycleAdapter to use RequestContext.getEvent() when handling the exceptions and as @pablokraan mentioned the MuleEvent in RequestContext is incorrect so if the flow has another component in catch-exception-strategy that access to it may incur in a problem.

What do you think would be the correct way to handle this?